edge-modules/MessagesAnalyzer/Settings.cs

```diff
@@ -36,17 +40,25 @@ class Settings
                     configuration.GetValue(DeviceIdPropertyName, DefaultDeviceId),
                     excludedModules,
                     configuration.GetValue(WebhostPortPropertyName, DefaultWebhostPort),
-                    configuration.GetValue(ToleranceInMillisecondsPropertyName, DefaultToleranceInMilliseconds));
+                    configuration.GetValue(ToleranceInMillisecondsPropertyName, DefaultToleranceInMilliseconds),
+                    configuration.GetValue<string>(LogAnalyticEnabledName),
+                    configuration.GetValue<string>(LogAnalyticWorkspaceIdName),
+                    configuration.GetValue<string>(LogAnalyticSharedKeyName),
+                    configuration.GetValue<string>(LogAnalyticLogTypeName));
             });
 
-        Settings(string eventHubCs, string consumerGroupId, string deviceId, IList<string> excludedModuleIds, string webhostPort, double tolerance)
+        Settings(string eventHubCs, string consumerGroupId, string deviceId, IList<string> excludedModuleIds, string webhostPort, double tolerance, string laEnabled, string laWorkspaceIdName, string laSharedKeyName, string laLogTypeName)
         {
             this.EventHubConnectionString = eventHubCs;
             this.ConsumerGroupId = consumerGroupId;
             this.ExcludedModuleIds = excludedModuleIds;
             this.DeviceId = deviceId;
             this.WebhostPort = webhostPort;
             this.ToleranceInMilliseconds = tolerance;
+            this.LogAnalyticEnabled = Convert.ToBoolean(laEnabled);
```

Comment:
this may throw exception; use Boolean.TryParse.

Reply:
Boolean.TryParse is added.